File: TelerikUIExamples/BindWithDelegate.m
//
//  BindWithDelegate.m
//  TelerikUIExamples
//
//  Copyright (c) 2013 Telerik. All rights reserved.
//

#import "BindWithDelegate.h"
#import <TelerikUI/TelerikUI.h>

@interface ChartDataSource : NSObject<TKChartDataSource>
@end

@implementation ChartDataSource

- (NSUInteger)numberOfSeriesForChart:(TKChart *)chart
{
    return 3;
}

- (TKChartSeries *)seriesForChart:(TKChart *)chart atIndex:(NSUInteger)index
{
    TKChartLineSeries *series = [[TKChartLineSeries alloc] init];
    series.selectionMode = TKChartSeriesSelectionModeDataPoint;
    series.style.pointShape = [[TKPredefinedShape alloc] initWithType:TKShapeTypeCircle andSize:CGSizeMake(10, 10)];
    series.selectionMode = TKChartSeriesSelectionModeSeries;
    series.title = [NSString stringWithFormat:@"Series %u", index+1];
    if (index == 2) {
        series.spline = YES;
    }
    return series;
}

- (NSUInteger)chart:(TKChart *)chart numberOfDataPointsForSeriesAtIndex:(NSUInteger)seriesIndex
{
    return 10;
}

- (id<TKChartData>)chart:(TKChart *)chart dataPointAtIndex:(NSUInteger)dataIndex forSeriesAtIndex:(NSUInteger)seriesIndex
{
    TKChartDataPoint *point = [[TKChartDataPoint alloc] init];
    point.dataXValue = @(dataIndex);
    point.dataYValue = @(arc4random() % 100);
    return point;
}

@end

@implementation BindWithDelegate
{
    TKChart *_chart;
    ChartDataSource *_chartDataSource;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.view.backgroundColor = [UIColor whiteColor];
    
    _chartDataSource = [[ChartDataSource alloc] init];
    
    _chart = [[TKChart alloc] initWithFrame:[self exampleBounds]];
    _chart.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    _chart.dataSource = _chartDataSource;
    _chart.legend.hidden = NO;
    _chart.legend.style.position = TKChartLegendPositionTop;
    _chart.legend.container.stack.orientation = TKStackLayoutOrientationHorizontal;
    [self.view addSubview:_chart];
}

@end

File: TelerikUIExamplesInCSharp/Examples/Chart/Binding/BindWithDelegate.cs
using System;
using System.Collections.Generic;

using MonoTouch.Foundation;
using MonoTouch.UIKit;

using TelerikUI;

namespace Examples
{
	public class BindingWithDelegate: ExampleViewController
	{
		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			TKChart chart = new TKChart (this.ExampleBounds);
			chart.AutoresizingMask = UIViewAutoresizing.FlexibleWidth | UIViewAutoresizing.FlexibleHeight;
			chart.DataSource = new ChartDataSource ();
			chart.Legend.Hidden = false;
			chart.Legend.Style.Position = TKChartLegendPosition.Top;
			chart.Legend.Container.Stack.Orientation = TKStackLayoutOrientation.Horizontal;
			this.View.AddSubview (chart);
		}

		class ChartDataSource: TKChartDataSource
		{
			Random r = new Random();

			public override uint NumberOfSeries (TKChart chart)
			{
				return 3;
			}

			public override TKChartSeries GetSeries (TKChart chart, uint index)
			{
				TKChartSeries series = null;

				if (index == 2) {
					series = new TKChartSplineSeries ();
				} else {
					series = new TKChartLineSeries ();
				}

				series.SelectionMode = TKChartSeriesSelectionMode.Series;
				series.Style.PointShape = new TKPredefinedShape (TKShapeType.Circle, new System.Drawing.SizeF (10, 10));
				series.Title = string.Format ("Series: {0}", index + 1);
				return series;
			}

			public override uint PointsInSeries (TKChart chart, uint seriesIndex)
			{
				return 10;
			}

			public override TKChartData GetPoint (TKChart chart, uint dataIndex, uint seriesIndex)
			{
				TKChartDataPoint point = new TKChartDataPoint ();
				point.DataXValue = new NSNumber (dataIndex);
				point.DataYValue = new NSNumber(r.Next (100));
				return point;
			}
		}
	}
}

